grease/src/Grease/Macaw/ResolveCall.hs

```diff
@@ -497,6 +500,58 @@ data LookupSyscallResult p sym arch atps rtps where
     Stubs.SomeSyscall p sym (Symbolic.MacawExt arch) ->
     LookupSyscallResult p sym arch atps rtps
 
+newtype SyscallHandleBuilder p sym arch
+  = SyscallHandleBuilder
+      ( forall atps rtps rtp blocks r ctx.
+        ArchContext arch ->
+        Int ->
+        Ctx.Assignment C.TypeRepr atps ->
+        Ctx.Assignment C.TypeRepr rtps ->
+        CS.CrucibleState
+          p
+          sym
+          (Symbolic.MacawExt arch)
+          rtp
+          blocks
+          r
+          ctx ->
+        LookupSyscallResult p sym arch atps rtps
+      )
+
+newtype PlatformContext arch
+  = PlatformContext
+  { macawSyscallLookup ::
+      forall bak sym p solver t st fs cExt.
+      ( OnlineSolverAndBackend solver sym bak t st fs
+      , HasGreaseSimulatorState p cExt sym arch
+      ) =>
+      bak ->
+      ArchContext arch ->
+      C.HandleAllocator ->
+      Symbolic.LookupSyscallHandle p sym arch
+  }
+
+stubsSyscallHandleBuilder ::
+  Map.Map
+    WFN.FunctionName
+    (Stubs.SomeSyscall p sym (Symbolic.MacawExt arch)) ->
+  SyscallHandleBuilder p sym arch
+stubsSyscallHandleBuilder syscallOvs =
+  SyscallHandleBuilder $ \arch syscallNum _ _ _ ->
+    case IntMap.lookup syscallNum (arch ^. archSyscallCodeMapping) of
+      Nothing ->
+        SkippedSyscall $ UnknownSyscallNumber syscallNum
+      Just syscallName ->
+        let syscallFnName = WFN.functionNameFromText syscallName
+         in case Map.lookup syscallFnName syscallOvs of
+              Just someSyscall ->
+                NewSyscall syscallName syscallNum someSyscall
+              Nothing ->
+                SkippedSyscall $ SyscallWithoutOverride syscallName syscallNum
+
+greaseSyscallFromSyscallBuilder :: GreaseLogAction -> ErrorSymbolicSyscalls -> (forall p sym. SyscallHandleBuilder p sym arch) -> PlatformContext arch
+greaseSyscallFromSyscallBuilder gla errSymbSyscalls sysBuilder = PlatformContext $ \bak arch halloc -> lookupSyscallHandle bak arch sysBuilder errSymbSyscalls (defaultLookupSyscallDispatch bak gla halloc arch)
```

Comment:
This name feels off, perhaps name it something to do with `PlatformContext`?

Also, could use a Haddock.